pkg/virt-api/rest/subresource.go

```diff
@@ -626,8 +627,24 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf(vmNotRunning)), response)
 			return
 		}
-		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v does not support manual stop requests", v1.RunStrategyHalted)), response)
-		return
+		if bodyStruct.GracePeriod == nil {
+			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v only supports manual stop requests with graceperiod=0", v1.RunStrategyHalted)), response)
+			return
+		}
+		if *bodyStruct.GracePeriod >= *vmi.Spec.TerminationGracePeriodSeconds {
```

Comment:
I think that here there is a possible `nil pointer dereference` if `vmi.Spec.TerminationGracePeriodSeconds` is nil. In this case I think we have to made a choice about how to consider if `bodyStruct.GracePeriod !=nil` and `vmi.Spec.TerminationGracePeriodSeconds == nil`, greater or shorter? Thanks

Reply:
I'm inclined to think that having any grace period would technically be shorter than not having one at all but let me know if that doesn't make sense to you.